### PR TITLE
VM Automation

### PR DESCRIPTION
**Pull request by tsegismont**
Automates the creation of the workshop VM with Packer.

The Fedora 26 VM comes with:

- Oracle JDK
- Maven
- Docker
- Openshift Client (oc)
- IntelliJ CE
- The workshop repo
- Maven dependencies and plugins downloaded

### CODE REVIEW

**Comment by tsegismont** [comment]
@galderz can you take a look ?

There are instructions for building the VM in the commit.

What's missing: Openshift Cluster setup (Infinispan templates included) and images download

**Comment by galderz** [comment]
Is NVM or Node.js? I use NVM locally but really what's needed here is node 4.2. In the current state of the workshop we need them to run the map UI... Eventually we might just dockerize them and depend on an image that brings what we need.

File: packer/README.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
+= Packer build for the Streaming Date Workshop Virtual Machine
+
+This build creates a Fedora 26 virtual machine having:
+
+- `root` with password `packer`
+- `developer` user with password `password` (_sudoer_)
```
Comment (galderz) [inline packer/README.adoc:6]: Following OpenShift's lead of using `developer`/`developer` as usr/pwd for entering the console, I've used that combination for the management access to data grid (for visualizer). Maybe we want to use the same combo here too?

File: packer/scripts/oc.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+DOCKER_BRIDGE=$(sudo docker network inspect -f "{{range .IPAM.Config }}{{ .Subnet }}{{end}}" bridge)
+
+sudo firewall-cmd --permanent --new-zone dockerc
+sudo firewall-cmd --permanent --zone dockerc --add-source ${DOCKER_BRIDGE}
+sudo firewall-cmd --permanent --zone dockerc --add-port 8443/tcp
+sudo firewall-cmd --permanent --zone dockerc --add-port 53/udp
+sudo firewall-cmd --permanent --zone dockerc --add-port 8053/udp
+sudo firewall-cmd --reload
+
+OC_VERSION=3.6.1
```
Comment (galderz) [inline packer/scripts/oc.sh:34]: Nice! I didn't know they had released a patch version... released 20h ago! :)

**Review by galderz — COMMENTED** [review]
I've not tested but looks good otherwise. I can give it a go tomorrow.

**Comment by galderz** [comment]
I briefly built it and tried it, and it looked good. I didn't run anything yet though. I want to focus on trying out the workshop after the modularization work.

**Comment by galderz** [comment]
@tsegismont I see you've added 2 commits. Anything else you need to add here?

**Comment by galderz** [comment]
@tsegismont Anything else to add? I'm planning to build the image this afternoon or tomorrow morning latest for duchess :|

**Comment by galderz** [comment]
@tsegismont I guess this is done now? I'll integrate later this morning and build an image...

File: packer/scripts/lab_setup.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+mvn install dependency:go-offline
+
+cd ${WORKSHOP_DIR}/web-viewer
+nvm use 4.2
+npm install
+npm run build
+
+cd ${WORKSHOP_DIR}
+./start-openshift.sh
+./start-datagrid.sh
```
Comment (galderz) [inline packer/scripts/lab_setup.sh:21]: I'm not sure about the role of this `lab_setup`, but `start-datagrid.sh` should not be executed by attendees. This is mostly there to start the datagrid via script, but attendees should do that via UI.

**Review by galderz — CHANGES_REQUESTED** [review]
The role of `lab_setup` script?

**Comment by galderz** [comment]
In fact, I don't think starting openshift and deploying components should be done on startup.

The idea we've been working with is that the attendees would first start openshift manually via script, create the datagrid, deploy the initial play with cache verticle, then deploy all components with their gaps in the code.

**Comment by galderz** [comment]
Maybe `lab_setup` is just a manual helper script for users to call? :|

I'm integrating it as is and I'll see how it goes.

**Pull request merged**